### PR TITLE
EPMKHRDMH-325/360 Mobile task list and task details

### PR DESCRIPTION
**Pull request by lotrien**
finished styling tasks list;
added controller for TasksList;
added service for percentage chart;

added controller for taskDetails;
finished styling tasks details;

### CODE REVIEW

File: mobile/modules/events/less/_events.details.less [hunk 1]
```diff
@@ -43,7 +43,7 @@
       display: flex;
       flex-flow: row nowrap;
       align-items: center;
-      padding: 1.4rem;
+      padding: 0.6rem 1.3rem;
       border-bottom: 1px solid @event-list-item-border-bottom;
       background-color: @color-white;
```
Comment (ghost) [inline mobile/modules/events/less/_events.details.less]: could you use component variable (mapped from color)

File: mobile/modules/events/less/_events.details.less [hunk 2]
```diff
@@ -60,11 +60,11 @@
...
     &__icons {
-      color: #000;
-      font-size: 2rem;
+      color: @event-list-item-text-color;
+      font-size: 2.5rem;
       padding-right: 3.2rem;
       &.dropp {
-        font-size: 1.4rem;
+        font-size: 2.5rem;
         color: #777;
```
Comment (ghost) [inline mobile/modules/events/less/_events.details.less]: could you use component variable (mapped from color)

File: mobile/modules/tasks/controllers/tasksDetails.client.controller.es6 [hunk 3]
```diff
@@ -10,17 +10,34 @@ class TasksDetailsController {
...
 
     showTask() {
-        this.task = this.Tasks.get({
+        this.Tasks.get({
             taskId: this.$stateParams.taskId
+        }).$promise.then((res) => {
+            this.task = res;
+            this.task.progress = this.TasksListService.recalcChart(this.task, '#172837', '#1AAA8F');
+            this.task.duration = moment(res.days.endTime).diff(moment(res.days.startTime), 'hours');
+            this.task.taskPriorityText = this.priorityForHuman(res);
```
Comment (ghost) [inline mobile/modules/tasks/controllers/tasksDetails.client.controller.es6]: could we take outside this logic (on promise then) to separate function?

File: mobile/modules/tasks/controllers/tasksList.client.controller.es6 [hunk 4]
```diff
@@ -36,43 +37,43 @@ class TasksListController {
...
+        this.setFiltersToLocalStorage();
+        break;
+      }
+      case 'isComplete': {
+        if (this.filter.isComplete) {
+          this.filteredTasks = this.tasks.filter(el => el.isComplete);
+        } else {
+          this.filteredTasks = this.tasks;
+        }
+        this.setFiltersToLocalStorage();
```
Comment (ghost) [inline mobile/modules/tasks/controllers/tasksList.client.controller.es6:65]: could we take outside logic for each of 'case' to separate function?

File: mobile/modules/tasks/controllers/tasksList.client.controller.es6 [hunk 5]
```diff
@@ -36,43 +37,43 @@ class TasksListController {
...
     }
   }
 
-
   find() {
     this.Tasks.query().$promise
         .then((resolved) => {
           this.tasks     = resolved;
-          this.tasksCopy = resolved.slice();
+          this.filteredTasks     = resolved;
```
Comment (ghost) [inline mobile/modules/tasks/controllers/tasksList.client.controller.es6]: could you name the params more intuitive (resolved - maybe tasks?)
why we created the copy of the tasks property? (tasks/filteredTasks)

File: mobile/modules/tasks/less/_task-details.less [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+      flex-flow: row nowrap;
+      align-items: center;
+      padding: 0.7rem 1rem;
+      border-bottom: 1px solid @event-list-item-border-bottom;
+      background-color: @color-white;
+      font-family: @font-open-sans-semibold;
+      & .icon-1 { color: @priority-1-color; }
+      & .icon-2 { color: @priority-2-color; }
+      & .icon-3 { color: @priority-3-color; }
+      & .icon-4 { color: @priority-4-color; }
```
Comment (ghost) [inline mobile/modules/tasks/less/_task-details.less:71]: we have classes for this case

_.item-priority-1_ and so on

File: mobile/modules/tasks/less/_task-list.less [hunk 7]
```diff
@@ -23,7 +23,7 @@
     display: flex;
     flex-flow: row nowrap;
     align-items: center;
-    padding: 0.5rem 1rem;
+    padding: 0 1rem 0 0;
     border-left: 0.5rem solid @event-list-item-color;
     border-bottom: 1px solid @event-list-item-border-bottom;
     background-color: @color-white;
```
Comment (ghost) [inline mobile/modules/tasks/less/_task-list.less]: could you use component variable (mapped from color)

File: mobile/modules/tasks/services/tasksList.client.service.es6 [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+                data:      [{
+                    value: progress.percent,
+                    label: ''
+                }, {
+                    value: 100 - progress.percent,
+                    label: ''
+                }],
+                colors:    [color1, color2],
+                formatter: function formatter() {
+                    return '';
```
Comment (ghost) [inline mobile/modules/tasks/services/tasksList.client.service.es6]: do we need formatter?

File: mobile/modules/tasks/views/list-tasks.client.view.html [hunk 9]
```diff
@@ -68,4 +67,8 @@ <h3 class="main-header-mobile__title">Tasks</h3>
         data-ng-hide="taskListCtrl.tasks.length">
         No tasks yet.<br/>Please, click on <span class="icon icon-add"></span> to add new one.
     </p>
+    <p class="tasks-list__no-items"
+       data-ng-show="taskListCtrl.tasks.length && taskListCtrl.filter.isComplete && !taskListCtrl.filteredTasks.length">
+        There are no completed tasks <br/>Please <button type="button" class="sidebar-button__reset" ng-click="taskListCtrl.resetCompleteFilter()">reset</button> filter to display all tasks.
```
Comment (ghost) [inline mobile/modules/tasks/views/list-tasks.client.view.html]: please use _data-_ prefix

**Review by ghost — CHANGES_REQUESTED** [review]

File: mobile/modules/tasks/controllers/tasksList.client.controller.es6 (hunk 5 above)
Comment (lotrien) [reply mobile/modules/tasks/controllers/tasksList.client.controller.es6]: filtered tasks were created to give an ability to filter completed/incomplete tasks; this was done previously in taskController for desktop app

File: mobile/modules/tasks/services/tasksList.client.service.es6 (hunk 8 above)
Comment (lotrien) [reply mobile/modules/tasks/services/tasksList.client.service.es6]: No, we don't really need it. removed. thanks

File: mobile/modules/tasks/views/list-tasks.client.view.html (hunk 9 above)
Comment (lotrien) [reply mobile/modules/tasks/views/list-tasks.client.view.html]: fixed

File: mobile/modules/tasks/less/_task-details.less (hunk 6 above)
Comment (lotrien) [reply mobile/modules/tasks/less/_task-details.less:71]: no, we have classes not for color itself, but for some specific rule for task list

File: mobile/modules/events/less/_events.details.less (hunk 2 above)
Comment (lotrien) [reply mobile/modules/events/less/_events.details.less]: yep, thanks

File: mobile/modules/tasks/less/_task-list.less (hunk 7 above)
Comment (lotrien) [reply mobile/modules/tasks/less/_task-list.less]: fixed

File: mobile/modules/events/less/_events.details.less (hunk 1 above)
Comment (lotrien) [reply mobile/modules/events/less/_events.details.less]: fixed

File: mobile/modules/tasks/controllers/tasksList.client.controller.es6 (hunk 5 above)
Comment (ghost) [reply mobile/modules/tasks/controllers/tasksList.client.controller.es6]: ok, thank you for the clarification

File: mobile/modules/tasks/less/_task-details.less (hunk 6 above)
Comment (ghost) [reply mobile/modules/tasks/less/_task-details.less:71]: ok, sorry for misunderstanding. we are going to update it next time (in scope of associated tickets #328, #329)

**Pull request merged**